go/common/sgx/pcs/tcb.go

```diff
@@ -158,35 +170,38 @@ type TCBInfo struct {
 	FMSPC                   string     `json:"fmspc"`
 	PCEID                   string     `json:"pceId"`
 	TCBType                 int        `json:"tcbType"`
-	TCBEvaluationDataNumber int        `json:"tcbEvaluationDataNumber"`
+	TCBEvaluationDataNumber uint32     `json:"tcbEvaluationDataNumber"`
 	TCBLevels               []TCBLevel `json:"tcbLevels"`
 }
 
-func (ti *TCBInfo) validate(ts time.Time) error {
+func (ti *TCBInfo) validate(ts time.Time, policy *QuotePolicy) error {
 	if ti.Version != requiredTCBInfoVersion {
 		return fmt.Errorf("pcs/tcb: unexpected TCB info version: %d", ti.Version)
 	}
 
 	// Validate TCB info is not expired/not yet valid based on current time.
 	var (
-		issueDate  time.Time
-		nextUpdate time.Time
-		err        error
+		issueDate time.Time
+		err       error
 	)
 	if issueDate, err = time.Parse(TimestampFormat, ti.IssueDate); err != nil {
 		return fmt.Errorf("pcs/tcb: invalid issue date: %w", err)
 	}
-	if nextUpdate, err = time.Parse(TimestampFormat, ti.NextUpdate); err != nil {
-		return fmt.Errorf("pcs/tcb: invalid issue date: %w", err)
+	if _, err = time.Parse(TimestampFormat, ti.NextUpdate); err != nil {
+		return fmt.Errorf("pcs/tcb: invalid next update date: %w", err)
 	}
 
 	if issueDate.After(ts) {
 		return fmt.Errorf("pcs/tcb: TCB info issue date in the future")
 	}
-	if nextUpdate.Before(ts) {
+	if ts.Sub(issueDate).Nanoseconds() > int64(policy.TCBValidityPeriod)*24*int64(time.Hour) {
```

Comment:
Comparing a difference expressed in ns, to an interval expressed in hours?

Reply:
[`time.Hour`](https://pkg.go.dev/time#Hour) is the number of nanoseconds in an hour, so this should be fine?

Reply:
Ok